File: otani/php/05.php
<?php
  $stdin = trim(fgets(STDIN));
  $stdin = explode(" ", $stdin);
  $str = $stdin[0];
  $num = $stdin[1];

  var_dump(ngram($str, $num));

  function ngram($str, $num){
    $ngram = array();
    $j = 0;
    for($i = 0; $i < (strlen($str) - $num + 1); $i++) {
        $ngram[$j++] = mb_substr($str, $i, $num);
    }
    return $ngram;
  }
?>

File: otani/python/1/05.py
from pprint import pprint

def Ngram(string, num):
    ngram = [0 for i in range(len(string) - num + 1)] # 初期化
    for i in range(len(string) - num + 1):
        ngram[i] = string[i:(i + num)]
    return ngram

string = "ketsugenokamisama"
ngram = Ngram(string, 3)
pprint(ngram)
